cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/RunnableTaskLauncher.java

```diff
@@ -35,10 +37,18 @@ public class RunnableTaskLauncher {
 
   public RunnableTaskLauncher(CConfiguration cConf,
       DiscoveryService discoveryService,
-      DiscoveryServiceClient discoveryServiceClient) {
+      DiscoveryServiceClient discoveryServiceClient,
+      MetricsCollectionService metricsCollectionService) {
     injector = Guice.createInjector(
         new ConfigModule(cConf),
-        new RunnableTaskModule(discoveryService, discoveryServiceClient)
+        new RunnableTaskModule(discoveryService, discoveryServiceClient),
```

Comment:
Please put it into RunnableTaskModule instead of creating a new anonymous one

Reply:
I thought of doing that initially, but system worker tasks are installing both RunnableTaskModule and MessagingClientModule. So adding this to RunnableTaskModule will cause multiple bindings for messaging service.

https://github.com/cdapio/cdap/blob/e81bb233fa54e757ace4bbd36a91eec48d967675/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/SystemAppTask.java#L164-L169

Reply:
Why messaging client is a problem? We are talking metrics collection here, no?

Reply:
You're right. I read MessagingClientModule and it registered as MetricsClientModule in my brain. I added MetricsCollectionServide to RunnableTaskModule.